supervisor/backups/backup.py

```diff
@@ -252,7 +258,27 @@ def data(self) -> dict[str, Any]:
 
     def __eq__(self, other: Any) -> bool:
         """Return true if backups have same metadata."""
-        return isinstance(other, Backup) and self._data == other._data
+        if not isinstance(other, Backup):
+            return False
+
+        # Compare all fields except ones about protection. Current encryption status does not affect equality
+        keys = self._data.keys() | other._data.keys()
+        for k in keys - {ATTR_PROTECTED, ATTR_CRYPTO}:
```

Comment:
Tests with an unencrypted/encrypted backup also made the `registries` setting to light up:

```
2025-01-23 11:56:19.845 DEBUG (MainThread) [supervisor.backups.backup] Backup 5ba8f1d7 and 5ba8f1d7 not equal because docker field has different value: {'registries': {'hub.docker.com': {'username': 'agners', 'password': '/aDMOAXyCd5fG1ZR2j4nHA=='}}} and {'registries': {'hub.docker.com': {'username': 'agners', 'password': 'gaga'}}}
2025-01-23 11:56:19.845 ERROR (MainThread) [supervisor.backups.manager] Ignoring backup 5ba8f1d7 in backup_test due to: Backup in backup_test and None both have slug 5ba8f1d7 but are not the same!
```

Technically we could dive into the field and only ignore the password (since this is the only part which is encrypted/unencrypted). However, the idea is to move the settings into a Supervisor specific tarball anyways, so I'd say we just ignore the whole attribute for now:

```suggestion
        for k in keys - {ATTR_PROTECTED, ATTR_CRYPTO, ATTR_DOCKER}:
```